service/worker/indexer/esProcessor.go

```diff
@@ -65,6 +67,7 @@ type (
 		config        *Config
 		logger        log.Logger
 		metricsClient metrics.Client
+		msgEncoder    codec.BinaryEncoder
```

Comment:
Making `esProcessorImpl` to have its own msgEncoder makes the code a little bit inflexible because if we need to change the message encoding in the further, we will have to modify both `esProcessor` and `indexerProcessor`. It also introduces some duplicated logic for message decoding. 

Given that `indexerProcessor` already has a msgEncoder and a method for decoding kafka message, can we somehow reuse that? For example, pass the encoder as a parameter when initializing `esProcessor`. For the decoding logic, maybe we can create a new util function for that.